scripts/update-sitemap.ts

```diff
@@ -0,0 +1,240 @@
+const linkedEventsUrl = process.env.LINKED_EVENTS_URL;
+const host = process.env.HOST_URL;
+const pageSize = 100;
+const pathToSitemaps: string = path.join(__dirname, "../public");
+
+const now = new Date();
+
+interface SimpleCollection {
+  id: string;
+  last_published_at: string;
+}
+
+interface SimpleEvent {
+  id: string;
+  last_modified_time: string;
+}
+
+/**
+ * Format date to the format that are expected for Google sitemaps
+ * @param {string} date
+ * @return {string}
+ */
+export const formatDate = (date: string): string =>
+  formatDateStr(new Date(date), "yyyy-DD-mm");
+
+/**
+ * Write object to a xml file
+ * @param {string} path
+ * @param {object} data
+ */
+const writeXMLFile = async (path: string, data: object) => {
+  const options = { compact: true, ignoreComment: true, spaces: 4 };
+  const xml = convert.js2xml(data, options);
+
+  await writeFile(path, xml, "utf8");
+};
+
+/**
+ * Fetch events from linkedevents
+ * @param {string} language
+ * @return {object[]}
+ */
+const getEvents = async (language: string) => {
+  const events: SimpleEvent[] = [];
+  let url =
+    `${linkedEventsUrl}/event` +
+    `?language=${language}` +
+    `&start=${now.toISOString()}` +
+    `&page_size=${pageSize}` +
+    "&division=kunta:helsinki" +
+    "&super_event_type=umbrella,none";
+
+  // Loop until linkedevents returns null to next page attribute
+  while (!!url) {
+    const start = new Date();
+    const res = await fetch(url);
+    const result = await res.json();
+
+    if (res.status !== 200) {
+      throw Error(
+        `Could not fetch ${language} events data from url ${url}: ${res.status} ${res.statusText}`
+      );
+    }
+
+    const end = new Date();
+    // eslint-disable-next-line no-console
+    console.log(`GET: ${url} (${end.getTime() - start.getTime()}ms)`);
+
+    const data = result.data;
+    events.push(...data);
+
+    url = result.meta.next;
+  }
+
+  return events;
+};
+
+/**
+ * Generate sitemap in xml format for events in selected language
+ * @param {object[]} events
+ * @param {string} language
+ */
+const generateEventSitemap = async (
+  events: SimpleEvent[],
+  language: string
+) => {
+  const data = {
+    _declaration: { _attributes: { encoding: "utf-8", version: "1.0" } },
+    urlset: {
+      _attributes: { xmlns: "http://www.sitemaps.org/schemas/sitemap/0.9" },
+      url: [
+        ...events.map(event => ({
+          lastmod: formatDate(event.last_modified_time),
+          loc: `${host}/${language}/event/${event.id}`
+        }))
+      ]
+    }
+  };
+
+  await writeXMLFile(`${pathToSitemaps}/sitemap_events_${language}.xml`, data);
+};
+
+/**
+ * Generate sitemap in xml format for events in all languages
+ */
+const generateEventSitemaps = async () => {
+  await Promise.all(
+    languages.map(async lang => {
+      try {
+        const events = await getEvents(lang);
+        // eslint-disable-next-line no-console
+        await generateEventSitemap(events, lang);
```

Comment:
Why do you need `await` here?
The callback you provided for `map` is returning `undefined` so you'll end up with an array of `undefined` passed to `Promise.all`